docs/dls/component-design-guidelines.md

```diff
@@ -0,0 +1,100 @@
+---
+id: component-design-guidelines
+title: Component Design Guidelines
+description: Documentation on Design
+---
+
+Be it a new component contribution, or plugin specific components, you'll want
+to follow these guidelines, we'll cover the 3 main subjects that define the
+general look and feel of your components, all of which build on top of the MUI
+theme features:
+
+- Layout
+- Color palette
+- Typography
+
+## 🏗️ Layout
+
+Layout refers to how you organize or stack content, whenever possible we want
+to use Backstage's components (check the [Storybook][1] for a list and demo)
+or MUI components (check the [MUI docs][2]).
+
+If none of these fit your layout needs, then you can build one, however using
+HTML+CSS directly is not recommended, it's better to use MUI layout components
+to make your layout theme aware, meaning if someone changes the theme, your
+layout would react to those changes without requiring updates to your code.
+
+Specifically you want to look at these components that make use of the
+`theme.spacing()` function for margins, paddings and positions, as well as
+color palette and typography:
+
+- [Container][3] mostly at page level
+- [Box][4] like a div that can be customized a lot
+- [Grid][5] & [Grid V2][6] (preferable V2) for flexible grid layouts
+- [Stack][7] for vertical layouts (single column)
+- [Paper][8] The base of a card, like it's background & padding on the borders
+- [Card][9] Card with support for title, description, buttons, images...
+
+## Color palette
+
+Any component that needs a color to put in the styles should be using the
+theme's color palette, most Backstage components and all MUI components should
+use the theme's color palette by default, so unless you need explicit control
+on the color of a component (say when the component was designed to use the
+primary color but you want to use the secondary), then the easiest way to access
+the color palette is with [`useTheme` hook](https://mui.com/material-ui/customization/theming/#accessing-the-theme-in-a-component).
+
+It's not a very common use case to override a theme color in a MUI component
+but let's say you have a Paper component that highlights it's content with a
+different color for a side menu or something (usually you use the elevation,
+but maybe the designer wanted a colorful app), you can use the theme like this:
+
+```tsx
+import { useTheme } from '@mui/material/styles';
+
+export function Sidebar() {
+  const theme = useTheme();
+  return (
+    <Paper
+      sx={{
```

Comment:
mui4 Paper doesn't have an `sx` prop - and we usually go via makeStyles and classes to adapt things. E.g.: https://github.com/backstage/backstage/blob/7539b3674830af1ca98ea3d74ca4be6b290c4473/packages/create-app/templates/default-app/packages/app/src/components/search/SearchPage.tsx#L55

Reply:
Indeed, that is the case for MUI 4, I'll change this to use `makeStyles` and classes instead.

Reply:
Done.